# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentChild.cpp
# Commit: 4cf9a1ce9e92
# Full Hash: 4cf9a1ce9e92dbcc022c1d8d7d6ac8ed185dc082
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 15:55:11
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 11 - Use RemoteBrowser interface instead of IPDL actors in nsFrameLoader. r=nika
#   
#   This commit replaces the direct use of the IPDL actors in nsFrameLoader with
#   the RemoteBrowser interface. Some special use cases are adapted to still use
#   the IPDL actors. In the future, we should burn these use cases down.
# ==============================================================================

diff -r 9d7151b26a9e -r 4cf9a1ce9e92 dom/ipc/ContentChild.cpp
--- a/dom/ipc/ContentChild.cpp	Wed May 15 10:34:03 2019 -0500
+++ b/dom/ipc/ContentChild.cpp	Mon May 06 21:29:48 2019 -0500
@@ -30,6 +30,7 @@
 #include "mozilla/docshell/OfflineCacheUpdateChild.h"
 #include "mozilla/dom/BrowsingContext.h"
 #include "mozilla/dom/BrowsingContextGroup.h"
+#include "mozilla/dom/BrowserBridgeHost.h"
 #include "mozilla/dom/ClientManager.h"
 #include "mozilla/dom/ClientOpenWindowOpActors.h"
 #include "mozilla/dom/ChildProcessMessageManager.h"
@@ -2007,7 +2008,7 @@
   csChild->TrackCookieLoad(aChannel);
 }
 
-already_AddRefed<BrowserBridgeChild> ContentChild::CreateBrowser(
+already_AddRefed<RemoteBrowser> ContentChild::CreateBrowser(
     nsFrameLoader* aFrameLoader, const TabContext& aContext,
     const nsString& aRemoteType, BrowsingContext* aBrowsingContext) {
   MOZ_ASSERT(XRE_IsContentProcess());
@@ -2054,7 +2055,9 @@
       aBrowsingContext, chromeFlags);
   browserBridge->mIPCOpen = true;
 
-  return browserBridge.forget();
+  RefPtr<BrowserBridgeHost> browserBridgeHost =
+      new BrowserBridgeHost(browserBridge);
+  return browserBridgeHost.forget();
 }
 
 PScriptCacheChild* ContentChild::AllocPScriptCacheChild(